src/HomeWork_4/HomeWork_4.ts

```diff
@@ -0,0 +1,41 @@
+// Створіть класи Circle, Rectangle, Square і Triangle. У кожного з них є загальнодоступний
+//  метод calculateArea.У кожної фігури є загальнодоступні властивості - колір і назва,
+// які не можна змінювати після створення.У Square і Rectangle зі свого боку є
+// ще додатковий метод print, який виводить рядок із формулою розрахунку площі
+
+interface IFigure {
+  readonly name: string;
+  readonly color: string;
+}
+
+class Circle implements IFigure {
+  constructor(public readonly name: string, public readonly color: string) {}
+  public calculateArea(radius: number): number {
```

Comment:
Please consider implementing an abstract class in this scenario. Let's observe the outcomes.